service/resource/legacyv2/adapter/instance.go

```diff
@@ -0,0 +1,116 @@
+package adapter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/keyv2"
+)
+
+// template related to this adapter: service/templates/cloudformation/instance.yaml
+
+type instanceAdapter struct {
+	MasterAZ                     string
```

Comment:
I've prefixed the AZ with Master but its needed in multiple adaptors. Should we have these shared fields in the main Adaptor struct?

Reply:
Yes, I think the shared fields should be promoted to the main Adapter, maybe the Master prefix still makes sense there.